### PR TITLE
Update the ethnic group and background questions in our hesa questionnaire

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

Update the ethnic group and background questions in our HESA questionnaire

## Guidance to review

1. Does it follow the prototype?

## Link to Trello card

https://trello.com/c/rQxPo0aD/729-needs-to-be-done-by-11-oct-update-the-ethnic-group-and-background-questions-in-our-hesa-questionnaire

### CODE REVIEW

**Comment by tomas-stefano** [comment]
[[Needs to be done by 11 Oct] Update the ethnic group and background questions in our HESA questionnaire](https://trello.com/c/rQxPo0aD/729-needs-to-be-done-by-11-oct-update-the-ethnic-group-and-background-questions-in-our-hesa-questionnaire)

**Comment by tomas-stefano** [comment]
For those helping to test, we need to test in the review app:

* A new application
* Carry over application with the ethnicity "Irish"
* Carry over application with any other ethnicity

And go and submit an application until it reaches the ethnic page or check your answers page.

**Review by elliotcm — APPROVED** [review]
Code looks good.

File: app/views/candidate_interface/equality_and_diversity/edit_ethnic_group.html.erb [hunk 1]
```diff
@@ -7,10 +7,13 @@
       <%= f.govuk_error_summary %>
 
       <%= f.govuk_radio_buttons_fieldset :ethnic_group, caption: { text: t('equality_and_diversity.title'), size: 'xl' }, legend: { text: t('equality_and_diversity.ethnic_group.title'), size: 'xl', tag: 'h1' } do %>
+        <p class="govuk-body">Training providers will not see your answer to this question when deciding whether to offer you a place.</p>
+        <p class="govuk-body">They will only see your answer if you accept an offer from them.</p>
+
```
Comment (frankieroberto) [inline app/views/candidate_interface/equality_and_diversity/edit_ethnic_group.html.erb:12]: The two paragraphs should come before the `govuk_radio_buttons_fieldset`, and a separate legend ("What is your ethnic group?") should be visible - separate from the h1 which is now just "Ethnicity". The caption would have to move too.

Something a bit like this:
```suggestion
        <p class="govuk-body">Training providers will not see your answer to this question when deciding whether to offer you a place.</p>
        <p class="govuk-body">They will only see your answer if you accept an offer from them.</p>
      <%= f.govuk_radio_buttons_fieldset :ethnic_group, legend: { text: "What is your ethnic group?", size: 'm' } do %>

```

File: app/views/candidate_interface/equality_and_diversity/edit_ethnic_group.html.erb [hunk 2]
```diff
@@ -7,10 +7,13 @@
       <%= f.govuk_error_summary %>
 
       <%= f.govuk_radio_buttons_fieldset :ethnic_group, caption: { text: t('equality_and_diversity.title'), size: 'xl' }, legend: { text: t('equality_and_diversity.ethnic_group.title'), size: 'xl', tag: 'h1' } do %>
```
Comment (frankieroberto) [inline app/views/candidate_interface/equality_and_diversity/edit_ethnic_group.html.erb:9]: I've dropped this heading size down from XL to L.

```suggestion
      <%= f.govuk_radio_buttons_fieldset :ethnic_group, caption: { text: t('equality_and_diversity.title'), size: 'xl' }, legend: { text: t('equality_and_diversity.ethnic_group.title'), size: 'l', tag: 'h1' } do %>
```

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Looking good - couple of changes requested.

Could you also update the Check your answers page so that it just shows the answer to the second question - eg "Roma" rather than using brackets "White (Roma)"? (That could be a second PR if needed).

File: spec/factories/application_form.rb [hunk 3]
```diff
@@ -58,13 +58,17 @@
 
     trait :with_equality_and_diversity_data do
       equality_and_diversity do
-        ethnicity = Class.new.extend(EthnicBackgroundHelper).all_combinations.sample
+        all_ethnicities = Class.new.extend(EthnicBackgroundHelper).all_combinations
+        if RecruitmentCycle.current_year < 2023
```
Comment (carlosmartinez) [inline spec/factories/application_form.rb:62]: might be good to have something like `Hesa::CODE_CHANGES_2023`

**Review by carlosmartinez — APPROVED** [review]

File: spec/factories/application_form.rb (hunk 3 above)
Comment (tomas-stefano) [reply spec/factories/application_form.rb:62]: I can do it but this is only used in specs to create certain data depending of the year. Should I still?

Comment (carlosmartinez) [reply spec/factories/application_form.rb:62]: i wouldn't bother if it were only used in this test, but i think it'd be useful in the disabilities PR also.

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**